br/pkg/lightning/restore/restore.go

```diff
@@ -2480,12 +2537,21 @@ func (cr *chunkRestore) encodeLoop(
 						}
 					}
 					filteredColumns = columnNames
+					extendCols = cr.chunk.FileMeta.ExtendData.Columns
+					extendColsMap := make(map[string]struct{})
+					for _, extendCol := range extendCols {
+						extendColsMap[extendCol] = struct{}{}
+					}
 					if ignoreColumns != nil && len(ignoreColumns.Columns) > 0 {
 						filteredColumns = make([]string, 0, len(columnNames))
 						ignoreColsMap := ignoreColumns.ColumnsMap()
+						for c := range ignoreColsMap {
+							delete(extendColsMap, c)
+						}
 						if len(columnNames) > 0 {
 							for _, c := range columnNames {
 								if _, ok := ignoreColsMap[c]; !ok {
+									delete(extendColsMap, c)
```

Comment:
When `columnNames` isn't given it means sql looks like
```sql
INSERT INTO t VALUES (1,1);
-- instead of INSERT INTO t (a,b) VALUES (1,1);
```
We can't tell from source data whether do users specify extend values or not. So I choose not to delete it, which means assign all values through our generation